### ISSUE TITLE
hsc2cs generated C files don't get linked into executables

### ISSUE THREAD

**Issue by edsko** [haskell/cabal#2538]
Given `.cabal` file

```
name:                test-package
version:             0.1.0.0
build-type:          Simple
cabal-version:       >=1.10

executable experiment
  main-is:             Main.hs
  other-modules:       SomeHscModule
  build-depends:       base >=4.7 && <4.9
  default-language:    Haskell2010
```

standard `Setup.hs`, `Main.hs` given by 

``` haskell
module Main where

import SomeHscModule

main :: IO ()
main = print valueOfFoo
```

and `SomeHscModule.hsc` given by

``` haskell
module SomeHscModule where

import Foreign.C
import System.IO.Unsafe

#def int cFoo() { return 1234; }

foreign import ccall "cFoo" cFoo :: IO CInt

valueOfFoo :: Int
valueOfFoo = unsafePerformIO $
  fromIntegral `fmap` cFoo
```

`cabal` creates the `.c` file correctly, but never compiles it or links it into the executable, leading to linker errors:

```
Undefined symbols for architecture x86_64:
  "_cFoo", referenced from:
      _r2rq_info in SomeHscModule.o
      _c2t2_info in SomeHscModule.o
     (maybe you meant: _SomeHscModule_cFoo_closure, _SomeHscModule_cFoo_info )
ld: symbol(s) not found for architecture x86_64
```

(tested with cabal 1.22 and HEAD, ghc 7.10.1, on OSX and on Ubuntu 64).

**Comment by dcoutts** [comment]
Or libs afaik. I don't think this (new?) hsc2hs feature has ever been supported by Cabal's build system. Patches welcome.

**Comment by ezyang** [comment]
Pretty sure this was fixed in a30b11fe74ec721043301f8de90b5e93efbb6359.
